examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+import type { DataSource, syncCollection } from "./data"
+
+import { ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { useState, useMemo, useLayoutEffect } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields } from "./data"
+import { Spinner } from "./components/Spinner"
+import { UI_DEFAULTS } from "./constants"
+
+interface FieldMappingRowProps {
+    originalField: ManagedCollectionField
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+function FieldMappingRow({ originalField, field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) {
+    const isUnsupported = !field
+    const hasFieldNameChanged = field!.name !== originalField.name
+    const fieldName = hasFieldNameChanged ? field!.name : ""
```

Comment:
If the field name hasn't changed we render an empty string, doesn't that mean that the value can suddenly disappear when writing the same word as the field has?

Let's say the field is already named _Cat_, and you want to add a suffix, _Toy_, now when you start typing _Cat_ the input is suddenly cleared. Haven't tested this, just think so by reading the code…

Reply:
Yeah true! I updated the code, it got simplified a lot